client/executor/wasmtime/src/instance_wrapper.rs

```diff
@@ -361,6 +361,33 @@ impl InstanceWrapper {
 						return;
 					}
 				}
+			} else if #[cfg(target_os = "macos")] {
+				use std::sync::Once;
+
+				unsafe {
+					let ptr = self.memory.data_ptr(&self.store);
+					let len = self.memory.data_size(&self.store);
+
+					// On MacOS we can simply overwrite memory mapping.
+					if libc::mmap(
+						ptr as _,
+						len,
+						libc::PROT_READ | libc::PROT_WRITE,
+						libc::MAP_FIXED | libc::MAP_PRIVATE | libc::MAP_ANONYMOUS,
+						-1,
+						0,
+					) == libc::MAP_FAILED {
+						static LOGGED: Once = Once::new();
+						LOGGED.call_once(|| {
+							log::warn!(
+								"WASM instance memory mmap failed: {}",
```

Comment:
`Failed to decommit WASM instance memory through mmap: {}` or something along those likes would be better I think, to make it explicit that the main purpose of this is not allocating new memory, but just clearing it.